### PR TITLE
Fix and add unit tests for issue 3071 - adaptive sampler runs on models w/ zero params.

### PR DESCRIPTION
**Pull request by mitzimorris**
#### Submission Checklist

- [x] Run unit tests: `./runTests.py src/test/unit`
- [x] Run cpplint: `make cpplint`
- [x] Declare copyright holder and open-source license: see below

#### Summary

PR https://github.com/stan-dev/stan/pull/3071 had a failing unit test; this PR fixes that test and adds a test to demonstrate that the adaptive sampler can be run on a model with zero parameters.

#### Intended Effect

Interfaces don't need to enforce calls to "fixed_param" sampler for models with zero params.

#### How to Verify

Unit tests

#### Side Effects

N/A

#### Documentation

N/A

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):  Columbia University

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: src/test/unit/services/util/inv_metric_test.cpp [hunk 1]
```diff
@@ -122,9 +122,11 @@ TEST(inv_metric, read_dense_OK) {
 
 TEST(inv_metric, read_dense_sz0) {
   stan::callbacks::logger logger;
-  stan::io::dump dmp = stan::services::util::create_unit_e_dense_inv_metric(0);
+  std::string txt = "inv_metric <- structure(c(), .Dim = c(0,0))";
+  std::stringstream in(txt);
+  stan::io::dump dump(in);
   Eigen::MatrixXd inv_inv_metric
-      = stan::services::util::read_dense_inv_metric(dmp, 0, logger);
+      = stan::services::util::read_dense_inv_metric(dump, 0, logger);
```
Comment (WardBrian) [inline src/test/unit/services/util/inv_metric_test.cpp]: Looking at the other tests in the file, this looks incorrect -- it should probably still be using create_unit_e_dense_inv_metric, just not using io::dump as the type. We're using `auto` elsewhere

Comment (mitzimorris) [reply src/test/unit/services/util/inv_metric_test.cpp]: I don't think this is the right test at all and it should be removed.
instead there should be a test `create_dense_sz0`, similar to existing test `create_diag_sz0`.     plus the test added by this PR.

Comment (mitzimorris) [reply src/test/unit/services/util/inv_metric_test.cpp]: changed this test, added test for create_dense_sz0

**Review by WardBrian — APPROVED** [review]

**Comment by SteveBronder** [comment]
This is causing a failure on jenkins in another pr in the math library.  Can we revert this, update it to develop, and open another pr?

https://jenkins.flatironinstitute.org/blue/organizations/jenkins/Stan%2FStan/detail/downstream_tests/573/pipeline

**Comment by WardBrian** [comment]
@SteveBronder that failure was from before this was merged, does it still happen now?
